source/dubregistry/dbcontroller.d

```diff
@@ -473,6 +473,36 @@ struct DbRepository {
 	string kind;
 	string owner;
 	string project;
+
+	void parseURL(URL url) {
+		string host = url.host;
+		if (!url.schema.among!("http", "https"))
+			throw new Exception("Invalid Repository Schema (only supports http and https)");
+		if (host.endsWith(".github.com") || host == "github.com" || host == "github") {
+			kind = "github";
+		} else if (host.endsWith(".gitlab.com") || host == "gitlab.com" || host == "gitlab") {
+			kind = "bitbucket";
+		} else if (host.endsWith(".bitbucket.org") || host == "bitbucket.org" || host == "bitbucket") {
+			kind = "gitlab";
+		} else {
+			throw new Exception("Please input a valid project URL to a GitHub, GitLab or BitBucket project.");
+		}
+		auto path = url.path.bySegment;
+		if (path.empty)
+			throw new Exception("Invalid Repository URL (no path)");
+		assert(path.front.name.empty); // first element is empty
```

Comment:
Could you put the comment as message?
(line number on stack traces is broken with newer glibc's and on OSX so the exact message is often helpful)

Reply:
Pushed the nit myself quickly.

Reply:
```
> git push ...
To github.com:WebFreak001/dub-registry
 ! [remote rejected] input-convenience (permission denied)
error: failed to push some refs to 'git@github.com:WebFreak001/dub-registry'
```

That's weird. I was able to squash it yesterday after all :O